src/main/java/Duke.java

```diff
@@ -1,10 +1,85 @@
+                } else {
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                }
+            } else if (userInput.contains("deadline")) {
+                String [] listArray = userInput.split("/",2);
+                String description = listArray[0];
+                String dueDate = listArray[1];
+                String info = description.substring(8).trim();
+                String due = dueDate.substring(3).trim();
+                tasks.add(new Deadline(info, due));
+                int finalTask = tasks.size() - 1;
+                System.out.println("Got it. I've added this task:");
+                System.out.println(tasks.get(finalTask).toString());
+                if (tasks.size() == 1) {
+                    System.out.println("Now you have " + tasks.size() + " task in the list.");
+                } else {
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                }
+            } else if (userInput.contains("event")) {
+                String [] listArray = userInput.split("/",3);
+                String description = listArray[0];
+                String startTime = listArray[1];
+                String endTime = listArray[2];
+                String info = description.substring(6).trim();
+                String start = startTime.substring(5).trim();
+                String end = endTime.substring(3).trim();
+                tasks.add(new Event(info, start, end));
+                int finalTask = tasks.size() - 1;
+                System.out.println("Got it. I've added this task:");
+                System.out.println(tasks.get(finalTask).toString());
+                if (tasks.size() == 1){
+                    System.out.println("Now you have " + tasks.size() + " task in the list.");
+                } else {
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                }
+            } else if(userInput.contains("mark")){
+                Integer itemNumber;
+                String [] commandIndex = userInput.split(" ",2);
+                itemNumber = Integer.parseInt(IndexArr[1])-1;
+                if (userInput.equals("unmark")) {
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    tasks.get((int) itemNumber).isCompleted = false;
+                    System.out.println(tasks.get(itemNumber).toString());
+                }
+                else {
+                    System.out.println("Nice! I've marked this task as done:");
+                    tasks.get(itemNumber).isCompleted = true;
+                    System.out.println(tasks.get(itemNumber).toString());
+                }
+            } else {
+                tasks.add(new Task(userInput, false));
+                System.out.println("added: " + userInput);
+            }
+        }
     }
-}
+}
```

Comment:
Relatively long and deep nested code. Consider break up the code with more abstractions. (create method for segments that are used repeatedly and a method for each command)